assets/js/compat.js

```diff
@@ -58,6 +58,32 @@ if (!Array.prototype.indexOf) {
   };
 }
 
+// From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Polyfill
```

Comment:
I think this shouldn't have been moved because `compat.js` only gets added for IEx-y versions but this shim is required for more than that.

Reply:
What about having two separate levels of compat?
Something like `compat-all.js` and `compat-ie.js`?

Reply:
Seems unnecessarily complicated. Just putting it into `util.js` with a comment as to why seems OK to me.

Reply:
Complicated? Seems perfectly reasonable to me ...
I mean, it doesn't really belong in `util.js` and neither in `init.js`, does it?

Reply:
It's unnecessarily complicated. If we start having multiple functions we polyfill for all browsers, then this makes sense. A file with literally one function is silly.

Reply:
Why don't we just put all polyfills inside one file in the first place?
I mean, it's not like we'd be shipping all too much unnecessarily.

Reply:
The app already ships more KB of JS than it should because we don't use proper minification. There's no point in making it worse than it currently is by giving everyone polyfills they don't need.

Reply:
Fine, but `util.js` is not ideal either. I'll move it back for now.

Reply:
It's not ideal but I think it's a good compromise between simplicity and performance.